FABulous/FABulous_API.py

```diff
@@ -293,3 +293,87 @@ def generateUserDesignTopWrapper(self, userDesign: Path, topWrapper: Path):
             Path to the output top wrapper file.
         """
         generateUserDesignTopWrapper(self.fabric, userDesign, topWrapper)
+
+    def genIOBelForTile(self, tile_name: str) -> list[Bel]:
+        """
+        Generates the IO BELs for the generative IOs of a tile.
+        Config Access Generative IOs will be a separate Bel.
+        Updates the tileDic with the generated IO BELs.
+
+        Parameters
+        ----------
+        tile_name : str
+            Name of the tile to generate IO Bels.
+
+        Returns
+        -------
+        bels : List[Bel]
+            The bel object representing the generative IOs.
+
+        Raises
+        ------
+        ValueError
+            If tile not found in fabric.
+            In case of an invalid IO type for generative IOs.
+            If the number of config access ports does not match the number of config bits.
+        """
+        tile = self.fabric.getTileByName(tile_name)
+        bels: list[Bel] = []
+        if not tile:
+            logger.error(f"Tile {tile_name} not found in fabric.")
```

Comment:
Can use the `logger.opt(exception=ValueError()).error(....)`, then you don't need to raise bellow